### PR TITLE
Adding structs to our integration tests

### PR DESCRIPTION
**Pull request by Tang8330**
Considering the following edge cases for structs:

1. Strings in Structs
2. Booleans in Structs
3. Numbers in Structs

### CODE REVIEW

File: integration_tests/destination_merge/main.go [hunk 1]
```diff
@@ -97,82 +93,8 @@ func (mt *MergeTest) verifyUpdatedData(numRows int) error {
...
-			if obj, ok := item.(map[string]interface{}); ok {
-				if field2, ok := obj["array_field2"].(float64); ok {
-					obj["array_field2"] = int(field2)
-				}
-			}
-		}
-
-		if !reflect.DeepEqual(expectedJSONArray, actualJSONArray) {
-			return fmt.Errorf("unexpected json_array for row %d: expected %v, got %v", i, expectedJSONArray, actualJSONArray)
+		if err := mt.framework.VerifyRowData(rows, i, 2.0); err != nil {
```
Comment (Tang8330) [inline integration_tests/destination_merge/main.go:96]: I consolidated this into one function

**Review by danafallon — APPROVED** [review]
